Source/DafnyCore/AST/Expressions/OldExpr.cs

```diff
@@ -14,13 +14,17 @@ public class OldExpr : Expression, ICloneable<OldExpr>, ICanFormat {
     Contract.Invariant(E != null);
   }
 
-  public OldExpr Clone(Cloner cloner) {
-    var result = new OldExpr(cloner.Tok(tok), cloner.CloneExpr(E), At);
+  public OldExpr(Cloner cloner, OldExpr original) : base(cloner, original) {
+    E = cloner.CloneExpr(original.E);
```

Comment:
As far as I can tell, the only difference with the previous code is that you are not cloning the token. Is cloning the token the thing that make it crash?

Reply:
Previously it _wasn't_ cloning the `Type` field, which is what led to the crash.